### PR TITLE
Content Processing Progress. Task 1 of 2

### PR DESCRIPTION
**Pull request by MichaelKovalchuk**
- Tracks content processing
- Improves error messaging for failed processes
- Add a randomized delay between progress calls

TODO (Task 2):
- Limit max number of items that can be processed at a time to 5

### CODE REVIEW

File: src/state/uploader.js [hunk 1]
```diff
@@ -12,34 +13,112 @@ export class Uploader {
...
+						contentId: content.id,
+						revisionId: revision.id,
+						percentComplete: progress.percentComplete,
+						ready: progress.ready
+					});
+					if (progress.ready) {
+						return;
+					}
+				} catch (error) {
+					if (error.status && error.status === 404) {
```
Comment (qwangd2l) [inline src/state/uploader.js]: Is it possible that we receive a different error status code and it'll end up monitoring indefinitely?

Comment (qwangd2l) [reply src/state/uploader.js]: Hmm or maybe it's just an intermittent error and we keep trying to get the progress.

Comment (MichaelKovalchuk) [reply src/state/uploader.js]: probably, i am copying this thing from content-service. my guess is that we may want to catch networking issues and the only call we are doing here is a progress check

**Comment by MichaelKovalchuk** [comment]
image

**Pull request merged**